packages/svelte/src/compiler/phases/3-transform/client/visitors/shared/utils.js

```diff
@@ -370,7 +370,7 @@ export function validate_mutation(node, context, expression) {
 export function build_expression(context, expression, metadata, state = context.state) {
 	const value = /** @type {Expression} */ (context.visit(expression, state));
 
-	if (context.state.analysis.runes) {
+	if (context.state.analysis.runes || context.state.analysis.maybe_runes) {
```

Comment:
```suggestion
	// Components not explicitly in legacy mode might be expected to be in runes mode (especially since we didn't
	// adjust this behavior until recently, which broke people's existing components), so we also bail in this case.
	// Kind of an in-between-mode.
	if (context.state.analysis.runes || context.state.analysis.maybe_runes) {
```